Content.Server/Antag/MobReplacementRuleSystem.cs

```diff
@@ -1,38 +1,179 @@
+                else
+                {
+                    BuildAMimicWorkshop(spawns[0].Entity, component);
+                }
+
+                if (k == MathF.MaxMagnitude(component.NumberToReplace, 1)
+                    && component.DoAnnouncement)
+                    _chat.DispatchStationAnnouncement(stations[0], Loc.GetString("station-event-rampant-intelligence-announcement"), playDefaultSound: true,
+                        colorOverride: Color.Red, sender: "Central Command");
+
+                k--;
+            }
+        }
+    }
+
+    /// <summary>
+    /// 	It's like Build a Bear, but MURDER
+    /// </summary>
+    /// <param name="uid"></param>
+    public void BuildAMimicWorkshop(EntityUid uid, MobReplacementRuleComponent component)
+    {
+        var metaData = MetaData(uid);
+        var vendorPrototype = metaData.EntityPrototype;
+        var mimicProto = _prototype.Index<EntityPrototype>(component.Proto);
+
+        var vendorComponents = vendorPrototype?.Components.Keys
+            .Where(n => n != "Transform" && n != "MetaData")
+            .Select(name => (name, _componentFactory.GetRegistration(name).Type))
+            .ToList() ?? new List<(string name, Type type)>();
 
-            Spawn(component.Proto, coordinates);
+        var mimicComponents = mimicProto?.Components.Keys
+            .Where(n => n != "Transform" && n != "MetaData")
+            .Select(name => (name, _componentFactory.GetRegistration(name).Type))
+            .ToList() ?? new List<(string name, Type type)>();
+
+        foreach (var name in mimicComponents.Except(vendorComponents))
+        {
+            var newComponent = _componentFactory.GetComponent(name.name);
+            EntityManager.AddComponent(uid, newComponent);
         }
+
+        var xform = Transform(uid);
+        if (xform.Anchored)
+            _transform.Unanchor(uid, xform);
+
+        SetupMimicNPC(uid, component);
+
+        if (TryComp<AdvertiseComponent>(uid, out var vendor)
+            && component.VendorModify)
+            SetupMimicVendor(uid, component, vendor);
+    }
+    /// <summary>
+    /// 	This handles getting the entity ready to be a hostile NPC
+    /// </summary>
+    /// <param name="uid"></param>
+    /// <param name="component"></param>
+    private void SetupMimicNPC(EntityUid uid, MobReplacementRuleComponent component)
+    {
+        _physics.SetBodyType(uid, BodyType.KinematicController);
+        _npcFaction.AddFaction(uid, "SimpleHostile");
+
+        var melee = EnsureComp<MeleeWeaponComponent>(uid);
+        melee.Angle = 0;
+        DamageSpecifier dspec = new()
+        {
+            DamageDict = new()
+            {
+                { "Blunt", component.MimicMeleeDamage }
+            }
+        };
+        melee.Damage = dspec;
+
+        var movementSpeed = EnsureComp<MovementSpeedModifierComponent>(uid);
+        (movementSpeed.BaseSprintSpeed, movementSpeed.BaseWalkSpeed) = (component.MimicMoveSpeed, component.MimicMoveSpeed);
```

Comment:
but why. No need to do it this way. It reduces readability A LOT
```suggestion
        movementSpeed.BaseSprintSpeed = component.MimicMoveSpeed;
        movementSpeed.BaseWalkSpeed = component.MimicMoveSpeed;
```